### ISSUE TITLE
No null checks for arrays of primitive types?

### ISSUE THREAD

**Issue by chrischu** [Singulink/RuntimeNullables#7]
I encountered a weird issue when using RuntimeNullables: apparently it does not generate null checks for primitive arrays (e.g. `byte[]`) as can be seen in the example below.

Is this expected behavior?

Source code:
```
namespace Namespace
{
  public class NullabilityTest
  {
    private readonly string _s;
    private readonly string[] _stringArray;
    private readonly byte[] _byteArray;
    private readonly int _i;

    public NullabilityTest (string s, string[] stringArray, byte[] byteArray, int i)
    {
      _s = s;
      _stringArray = stringArray;
      _byteArray = byteArray;
      _i = i;
    }
  }
}
```

IL:
```
.method public hidebysig specialname rtspecialname instance void
    .ctor(
      string s,
      string[] stringArray,
      unsigned int8[] byteArray,
      int32 i
    ) cil managed
  {
    .maxstack 2

    // [10 5 - 10 85]
    IL_0000: ldarg.1      // s
    IL_0001: brtrue.s     IL_000d
    IL_0003: ldstr        "s"
    IL_0008: call         void RuntimeNullables.ThrowHelpers::ThrowArgumentNull(string)
    IL_000d: ldarg.2      // stringArray
    IL_000e: brtrue.s     IL_001a
    IL_0010: ldstr        "stringArray"
    IL_0015: call         void RuntimeNullables.ThrowHelpers::ThrowArgumentNull(string)
    IL_001a: ldarg.0      // this
    IL_001b: call         instance void [mscorlib]System.Object::.ctor()
    IL_0020: nop

    // [11 5 - 11 6]
    IL_0021: nop

    // [12 7 - 12 14]
    IL_0022: ldarg.0      // this
    IL_0023: ldarg.1      // s
    IL_0024: stfld        string SignPath.Application.Domain.Common.NullabilityTest::_s

    // [13 7 - 13 34]
    IL_0029: ldarg.0      // this
    IL_002a: ldarg.2      // stringArray
    IL_002b: stfld        string[] SignPath.Application.Domain.Common.NullabilityTest::_stringArray

    // [14 7 - 14 30]
    IL_0030: ldarg.0      // this
    IL_0031: ldarg.3      // byteArray
    IL_0032: stfld        unsigned int8[] SignPath.Application.Domain.Common.NullabilityTest::_byteArray

    // [15 7 - 15 14]
    IL_0037: ldarg.0      // this
    IL_0038: ldarg.s      i
    IL_003a: stfld        int32 SignPath.Application.Domain.Common.NullabilityTest::_i

    // [16 5 - 16 6]
    IL_003f: ret

  } // end of method NullabilityTest::.ctor
```

**Comment by mikernet** [comment]
That's not intended behavior - I'll have a look. Thanks for the report :)

**Comment by mikernet** [comment]
Fixed in v1.0.2 via https://github.com/Singulink/RuntimeNullables/commit/5942115f3aafe3fa96417d8bef5402f9fade7130
